### PR TITLE
A note about converters Priority

### PR DESCRIPTION
**Pull request by sarxos**
Hello @dmlloyd. This is some brief documentation update in regards to custom converters priority we discussed in my PR.

### CODE REVIEW

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 1]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
 MicroProfileCustomValue value;
 ----
 
+==== Converter priority
+
+In some cases you may want to use custom converter to convert type which is already converted
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
In some cases you may want to use a custom converter to convert a type which is already converted
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 2]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+by a different converter. In such case you can use `javax.annotation.Priority` annotation to
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
by a different converter. In such cases, you can use the `javax.annotation.Priority` annotation to
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 3]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+change converters precedence and make your custom converter of higher priority then the other
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
change converters precedence and make your custom converter of higher priority than the other
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 4]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+on the list.
+
+By default, if no `@Priority` can be found on a converter, it's registered with priority of 100
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
By default, if no `@Priority` can be found on a converter, it's registered with a priority of 100
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 5]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+and all {project-name} core converters are registered with priority of 200, so depending on which
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
and all {project-name} core converters are registered with a priority of 200, so depending on which
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 6]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+converter you would like to replace, you need to set higher value.
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
converter you would like to replace, you need to set a higher value.
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 7]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+
+To demonstrate the idea let us implement custom converter which will take precedence over
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
To demonstrate the idea let us implement a custom converter which will take precedence over 
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 8]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+by a different converter. In such case you can use `javax.annotation.Priority` annotation to
+change converters precedence and make your custom converter of higher priority then the other
+on the list.
+
+By default, if no `@Priority` can be found on a converter, it's registered with priority of 100
+and all {project-name} core converters are registered with priority of 200, so depending on which
+converter you would like to replace, you need to set higher value.
+
+To demonstrate the idea let us implement custom converter which will take precedence over 
+`MicroProfileCustomValueConverter` implemented in previous example.
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]:
```suggestion
`MicroProfileCustomValueConverter` implemented in the previous example.
```

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 9]
```diff
@@ -258,6 +258,51 @@ After this is done you can use your custom type as a configuration value:
...
+        return new MicroProfileCustomValue(secretNumber);
+    }
+}
+----
+
+Since it converts the same value type (namely `MicroProfileCustomValue`) and has a priority
+of 150, it will be used instead of a `MicroProfileCustomValueConverter` which has a default
+priority of 100.
+
+NOTE: This new converter also needs to be listed in a service file.
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]: Maybe you can name the service file?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a few suggestion inline.

Let's wait for @dmlloyd's review.

File: docs/src/main/asciidoc/application-configuration-guide.adoc (hunk 9 above)
Comment (sarxos) [reply docs/src/main/asciidoc/application-configuration-guide.adoc]: Hi @gsmet, thank you for inspecting this. All requested changes are now addressed.

**Comment by sarxos** [comment]
Hello @gsmet, 

Thank you for inspecting this! All changes you requested are now addressed in a subsequent commit.

**Review by dmlloyd — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**